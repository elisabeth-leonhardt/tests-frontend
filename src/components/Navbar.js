import React from "react";

function Navbar({ user, setUser }) {
  return (
    <nav className="bg-dark-background">
      <div className="max-w-6xl p-4 mx-auto text-white flex justify-between items-center border-1 border-rose-500">
        <img src="bitlogic.png" alt="bitlogic logo" className="w-40" />
        <div className="flex gap-6">
          <a href="/" className="flex gap-2 hover:text-bitlogic-blue">
            <svg
              xmlns="http://www.w3.org/2000/svg"
              className="h-6 w-6"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
              strokeWidth={2}
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6"
              />
            </svg>
            Home
          </a>
          {user.length > 0 && (
            <a
              href={`/usermemes?username=${user}`}
              className="flex gap-2 hover:text-bitlogic-blue"
            >
              <svg
                xmlns="http://www.w3.org/2000/svg"
                className="h-6 w-6"
                fill="none"
                viewBox="0 0 24 24"
                stroke="currentColor"
                strokeWidth={2}
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  d="M14.828 14.828a4 4 0 01-5.656 0M9 10h.01M15 10h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                />
              </svg>
              Mis Memes
            </a>
          )}
        </div>
        {user.length > 0 ? (
          <button className="text-bitlogic-yellow">{user}</button>
        ) : (
          <butto>Login</butto>
        )}
      </div>
    </nav>
  );
}

export default Navbar;
