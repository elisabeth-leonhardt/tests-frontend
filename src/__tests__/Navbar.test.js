import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Navbar from "../components/Navbar";

describe("validar el componente navbar", () => {
  it("renderiza correctamente sin usuario", () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar user='' setUser={() => console.log("setting user")}></Navbar>
      </BrowserRouter>
    );
    expect(screen.getByText("Home")).toBeInTheDocument();
    expect(screen.getByText("Login")).toBeInTheDocument();
    expect(screen.queryByText("Mis Memes")).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renderiza correctamente con usuario", () => {
    render(
      <BrowserRouter>
        <Navbar
          user='funny-eli'
          //   setUser={() => console.log("setting user")}
        ></Navbar>
      </BrowserRouter>
    );
    screen.debug();
    expect(screen.getByText("Home")).toBeInTheDocument();
    expect(screen.getByText("funny-eli")).toBeInTheDocument();
    expect(screen.queryByText("Login")).not.toBeInTheDocument();
  });
  it("renderiza correctamente con usuario", () => {
    render(
      <BrowserRouter>
        <Navbar
          user='funny-eli'
          //   setUser={() => console.log("setting user")}
        ></Navbar>
      </BrowserRouter>
    );
    screen.debug();
    expect(screen.getByText("Home")).toBeInTheDocument();
    expect(screen.getByText("funny-eli")).toBeInTheDocument();
    expect(screen.queryByText("Login")).not.toBeInTheDocument();
  });
});
