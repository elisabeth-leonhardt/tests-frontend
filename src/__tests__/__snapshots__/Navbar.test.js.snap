// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validar el componente navbar renderiza correctamente sin usuario 1`] = `
<div>
  <aside
    class="bg-dark-background"
  >
    <div
      class="max-w-6xl p-2 mx-auto text-white flex justify-between items-center border-1 border-rose-500"
    >
      <img
        alt="bitlogic logo"
        class="w-40"
        src="BITMEMES.png"
      />
      <div
        class="flex gap-6"
      >
        <a
          class="flex gap-2 hover:text-bitlogic-blue"
          href="/"
        >
          <svg
            class="h-6 w-6"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          Home
        </a>
      </div>
      <button>
        Login
      </button>
    </div>
  </aside>
</div>
`;
